backend/infrahub/core/query/node.py

```diff
@@ -939,6 +932,16 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
             await self._add_node_order_attributes(
                 db=db, field_attribute_requirements=field_attribute_requirements, branch_filter=branch_filter
             )
+            for far in field_attribute_requirements:
+                if not far.is_order:
+                    continue
+                if far.supports_profile:
+                    self.order_by.append(far.final_value_query_variable)
+                    continue
+                self.order_by.append(far.node_value_query_variable)
```

Comment:
I let this block within `should_order` branch condition as this block seems only related to ordering.